### PR TITLE
build documentation using xmosdoc

### PR DESCRIPTION
**Pull request by shuchitak**
Gets the documentation building with xmosdoc.

### CODE REVIEW

File: doc/rst/app_316_mc.rst [hunk 1]
```diff
@@ -80,55 +81,55 @@ The master clock source is controlled by a mux which, in turn, is controlled by
...
      - 22.579MHz
 
 DAC and ADC Configuration
-+++++++++++++++++++++++++
+-------------------------
 
 The board is equipped with a single multi-channel audio DAC (Cirrus Logic CS4384) and a single
-multi-channel ADC (Cirrus Logic CS5368) giving 8 channels of analogue output and 8 channels of 
+multi-channel ADC (Cirrus Logic CS5368) giving 8 channels of analogue output and 8 channels of
 analogue input.
```
Comment (danielpieczko) [inline doc/rst/app_316_mc.rst:110]: These part numbers are not correct for the XK-AUDIO-316-MC board. There are four stereo DACs (TI PCM5122) and two 4-channel ADCs (TI PCM1865).

**Review by danielpieczko — APPROVED** [review]
Some comments/errors, but overall looks good.

File: README.rst [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
-:Vendor: XMOS
+sw_usb_audio: USB Audio reference designs
+#########################################
 
-Please note, Alpha and Beta releases may not accurately reflect the final release and documentation may not be complete. These early releases are not suitable for a production context, and are provided for evaluation purposes only. See 'Release Quality & QA'.
+:vendor: XMOS
+:version: 9.0.0
+:scope: General Use
+:description: USB Audio reference designs
+:category: General Purpose
```
Comment (xross) [inline README.rst:10]:
```suggestion
:category: Audio

File: doc/rst/drivers.rst [hunk 3]
```diff
@@ -35,7 +38,6 @@ feature-set/benefits:
     - Supports Windows sound APIs such as WASAPI, DirectSound, MME
     - Supports high-end audio features such as bit-perfect PCM up to 768 kHz sampling rate, native DSD format (through ASIO) up to DSD1024
     - Supports multiple clock sources such as S/PDIF, ADAT or WCLK inputs
-    - Supports MIDI 1.0 class, including MIDI port sharing
```
Comment (xross) [inline doc/rst/drivers.rst:38]: They still support his, we just don't use it for testing

File: doc/rst/app_316_mc.rst (hunk 1 above)
Comment (shuchitak) [reply doc/rst/app_316_mc.rst]: Thanks! fixed it.

File: tests/test_version.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+            if bcd_version['BCD_DEVICE_N'] != settings_yml_version[2]:
+                print(f"Error: BCD_DEVICE_N between version.h ({bcd_version['BCD_DEVICE_N']}) and settings.yml ({settings_yml_version[2]}) doesn't match")
+                fail = True
+
+            assert fail == False, f"Version mismatch between settings.yml ({settings_yml_version}) and version.h ({bcd_version})"
+
+if __name__ == "__main__":
+    test_version()
+
```
Comment (xross) [inline tests/test_version.py:42]: :)

**Review by xross — APPROVED** [review]

File: Jenkinsfile [hunk 5]
```diff
@@ -96,77 +101,67 @@ pipeline {
...
+                    runSwrefChecks("${WORKSPACE}/${REPO}", "v2.0.1")
+                  } // warnError("libchecks")
+                } // withTools("${env.TOOLS_VERSION}")
+              } // steps
+            } // stage('Library checks')
+
+            stage('Build Documentation') {
+              steps {
+                dir("${REPO}") {
+                  sh 'ls -l ..'
```
[the remark this answers is not in the document]
Comment (shuchitak) [reply Jenkinsfile]: done

File: README.rst [hunk 6]
```diff
@@ -1,16 +1,41 @@
-XMOS USB Audio 2.0 Reference Design README
-##########################################
+:orphan:
 
-:Version: 8.1.0
-:Vendor: XMOS
+sw_usb_audio: USB Audio reference designs
+#########################################
```
[the remark this answers is not in the document]
Comment (shuchitak) [reply README.rst:4]: fixed

File: doc/rst/drivers.rst (hunk 3 above)
Comment (shuchitak) [reply doc/rst/drivers.rst:38]: fixed it.

**Comment by xross** [comment]
I note the changelog checker has added "Added dependency" rather than "updated dependency" 

I think this is due to using "Legacy release history"

**Comment by xross** [comment]
Merging this with a view to fix changelog manually

**Pull request merged**